db/error_handler.cc

```diff
@@ -209,6 +225,11 @@ Status ErrorHandler::SetBGError(const Status& bg_err, BackgroundErrorReason reas
     auto_recovery = false;
   }
 
+  if (BackgroundErrorReason::kManifestWrite == reason) {
```

Comment:
Should this be done in the block in lines 282-296 as well?

Reply:
Make sense.
Some more thought: even if the error is a data loss w.r.t to the current MANIFEST, the database can still be recovered if next write to a new MANIFEST succeeds. With this being said, I think we can do this in one shot in line 272.

Reply:
Make sense.
Some more thought: even if the error is a data loss w.r.t to the current MANIFEST, the database can still be recovered if next write to a new MANIFEST succeeds. With this being said, I think we can do this in one shot in line 272.